org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/impl/BuilderStateDiscarder.java

```diff
@@ -39,13 +45,41 @@ public boolean forgetLastBuildState(Iterable<IProject> toUpdate, Map<String, Str
 				XtextBuilder builder = BuildManagerAccess.findBuilder(project);
 				if (builder != null) {
 					builder.forgetLastBuiltState();
+					touchProject(project);
 				}
 			}
 			return true;
 		}
 		return false;
 	}
 
+	/**
+	 * Touch a given project such that the auto-build knows it should be rebuild.
+	 *
+	 * @since 2.26
+	 */
+	protected void touchProject(IProject project) {
+		if (!project.getWorkspace().isTreeLocked()) {
+			try {
+				project.touch(null);
+			} catch (CoreException e) {
+				logger.error("Failed to refresh project while forgetting its builder state", e);
+			}
+		} else {
+			Job touchJob = Job.create("Touch project " + project.getName(), progressMonitor -> {
+				try {
+					project.touch(progressMonitor);
+					return Status.OK_STATUS;
+				} catch (CoreException e) {
+					logger.error("Failed to refresh project while forgetting its builder state", e);
+					return Status.CANCEL_STATUS;
+				}
+			});
+			touchJob.setRule(project);
```

Comment:
do we have to set userfalse/system?

Reply:
I think users would be happy to see that a build is scheduled/running? I can set system and user to false if you want

Reply:
Well I would be setting `user` to `false` so that should be okay right?

Reply:
i asume user=false and system=true
@szarnekow what do you think?

Reply:
👍